diff --git a/CMakeLists.txt b/CMakeLists.txt
index dae638be8..dd5d17794 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,6 +60,7 @@ option(USE_SHARED_CURL    "Use your installed copy of curl" off)
 option(USE_SHARED_GIFLIB  "Use your installed copy of giflib" off)
 option(USE_SHARED_JPEGLIB "Use your installed copy of jpeglib" off)
 option(USE_SHARED_ZLIB    "Use your installed copy of zlib" off)
+option(USE_SHARED_LIBARCHIVE "Use your installed copy of libarchive" off)
 option(USE_SHARED_LIBPNG  "Use your installed copy of libpng" off)
 option(USE_SHARED_TINYXML "Use your installed copy of tinyxml" off)
 option(USE_SHARED_PIXMAN  "Use your installed copy of pixman" off)
@@ -202,6 +202,15 @@ else()
 endif()
 include_directories(${ZLIB_INCLUDE_DIRS})
 
+# libarchive
+if(USE_SHARED_LIBARCHIVE)
+  find_library(ARCHIVE_LIBRARIES NAMES archive)
+  find_path(ARCHIVE_INCLUDE_DIRS NAMES archive.h)
+else()
+  add_definitions(-DLIBARCHIVE_STATIC)
+  set(ARCHIVE_LIBRARIES archive_static)
+endif()
+
 # libpng
 if(USE_SHARED_LIBPNG)
   find_package(PNG REQUIRED)
diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index a9e80d73c..619acac92 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -91,9 +91,6 @@ if(WITH_WEBP_SUPPORT)
   add_definitions(-DASEPRITE_WITH_WEBP_SUPPORT)
 endif()
 
-# libarchive definitions
-add_definitions(-DLIBARCHIVE_STATIC)
-
 ######################################################################
 # app-lib target
 
@@ -551,7 +548,7 @@ target_link_libraries(app-lib
   ${FREETYPE_LIBRARIES}
   ${HARFBUZZ_LIBRARIES}
   json11
-  archive_static
+  ${ARCHIVE_LIBRARIES}
   fmt
   tinyexpr)
 
diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index 596ffa2e4..977816439 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -105,16 +105,18 @@ endif()
 add_subdirectory(json11)
 
 # libarchive
-set(ENABLE_WERROR OFF CACHE BOOL "Treat warnings as errors - default is ON for Debug, OFF otherwise.")
-set(ENABLE_TEST OFF CACHE BOOL "Enable unit and regression tests")
-set(ENABLE_COVERAGE OFF CACHE BOOL "Enable code coverage (GCC only, automatically sets ENABLE_TEST to ON)")
-set(ENABLE_LZMA OFF CACHE BOOL "Enable the use of the system LZMA library if found")
-set(ENABLE_CNG OFF CACHE BOOL "Enable the use of CNG(Crypto Next Generation)")
-set(ENABLE_BZip2 OFF CACHE BOOL "Enable the use of the system BZip2 library if found")
-set(ENABLE_LIBXML2 OFF CACHE BOOL "Enable the use of the system libxml2 library if found")
-add_subdirectory(libarchive)
-target_include_directories(archive_static INTERFACE
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libarchive/libarchive>)
+if(NOT USE_SHARED_LIBARCHIVE)
+  set(ENABLE_WERROR OFF CACHE BOOL "Treat warnings as errors - default is ON for Debug, OFF otherwise.")
+  set(ENABLE_TEST OFF CACHE BOOL "Enable unit and regression tests")
+  set(ENABLE_COVERAGE OFF CACHE BOOL "Enable code coverage (GCC only, automatically sets ENABLE_TEST to ON)")
+  set(ENABLE_LZMA OFF CACHE BOOL "Enable the use of the system LZMA library if found")
+  set(ENABLE_CNG OFF CACHE BOOL "Enable the use of CNG(Crypto Next Generation)")
+  set(ENABLE_BZip2 OFF CACHE BOOL "Enable the use of the system BZip2 library if found")
+  set(ENABLE_LIBXML2 OFF CACHE BOOL "Enable the use of the system libxml2 library if found")
+  add_subdirectory(libarchive)
+  target_include_directories(archive_static INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libarchive/libarchive>)
+endif()
 
 # benchmark
 if(ENABLE_BENCHMARKS)
