diff --git a/CMakeLists.txt b/CMakeLists.txt
index 351603a..84afcf5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 IF(BIICODE)
     INCLUDE(biicode/cmake/tools)
-    
+
     ACTIVATE_CPP11()
     ADD_BIICODE_TARGETS()
 ELSE()
@@ -23,6 +23,9 @@ ELSE()
     include_directories(${CURL_INCLUDE_DIRS}
       ${CURLCPP_SOURCE_DIR}/include)
 
+    file(GLOB HEADER_LIST "${CMAKE_SOURCE_DIR}/include/*.h")
+    install (FILES ${HEADER_LIST} DESTINATION include)
+
     # Set up source directories
     add_subdirectory(src)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 56e4de1..13d50af 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,5 +17,6 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
 endif()
 
-add_library(curlcpp STATIC ${CURLCPP_SOURCE})
+add_library(curlcpp SHARED ${CURLCPP_SOURCE})
 target_link_libraries(curlcpp ${CURL_LIBRARY})
+install (TARGETS curlcpp LIBRARY DESTINATION lib)
